libs/user-facing-errors/src/query_engine.rs

```diff
@@ -148,3 +148,75 @@ pub struct RawQueryFailed {
     pub code: String,
     pub message: String,
 }
+
+#[derive(Debug, UserFacingError, Serialize)]
+#[user_facing(code = "P2011", message = "Null constraint violation on the ${constraint}")]
+pub struct NullConstraintViolation {
+    pub constraint: DatabaseConstraint,
+}
+
```

Comment:
Can the user tell which model /field was affected from the constraint?

Reply:
Sadly no, I haven't managed to extract more context from the error, and a discussion. I talked about it with Dom and it's pretty complicated, probably something we should do in another PR, if I understood right.